src/shims/x86/sse.rs

```diff
@@ -518,27 +521,36 @@ fn unary_op_f32<'tcx>(
         }
         FloatUnaryOp::Rcp => {
             let op = op.to_scalar();
-            // Use 1.0001 as a crude way of simulating inaccuracy of rcp
-            // 1.0001 / x ~= 1 / x
-            let one = Single::from_bits(1.0001f32.to_bits().into());
-            // floating point operation without side effects
-            #[allow(clippy::arithmetic_side_effects)]
+            // Calculate an approximation of 1 / x as (1 + err) / x, where
+            // err is a random value with a magnitude in the order of 2^-12
+            let one = (Single::from_u128(1).value + gen_random_float_error(this, -12)).value;
             Ok(Scalar::from_f32((one / op.to_f32()?).value))
         }
         FloatUnaryOp::Rsqrt => {
             let op = op.to_scalar();
+            // Calculate an approximation of 1 / sqrt(x) as (1 + err) / sqrt(x), where
+            // err is a random value with a magnitude in the order of 2^-12
+            let one = (Single::from_u128(1).value + gen_random_float_error(this, -12)).value;
             // FIXME using host floats
             let sqrt = Single::from_bits(f32::from_bits(op.to_u32()?).sqrt().to_bits().into());
-            // Use 1.0001 as a crude way of simulating inaccuracy of rsqrt
-            // 1.0001 / sqrt(x) ~= 1 / sqrt(x)
-            let one = Single::from_bits(1.0001f32.to_bits().into());
-            // floating point operation without side effects
-            #[allow(clippy::arithmetic_side_effects)]
             Ok(Scalar::from_f32((one / sqrt).value))
         }
     }
 }
 
+/// Returns a random a value in the range (-2^scale, 2^scale)
+fn gen_random_float_error<F: rustc_apfloat::Float>(
+    this: &mut crate::MiriInterpCx<'_, '_>,
+    scale: i32,
+) -> F {
+    let rng = this.machine.rng.get_mut();
+    // generates rand(0, 2^64) * 2^(scale - 64) = rand(0, 1) * 2^scale
+    let v = F::from_u128(rng.gen::<u64>().into()).value.scalbn(scale.checked_sub(64).unwrap());
+    // give it a random sign
+    #[allow(clippy::arithmetic_side_effects)] // floating point operation without side effects
+    if rng.gen::<bool>() { -v } else { v }
```

Comment:
That clippy lint triggers on the float types...?!? This is an overloaded operator, I am quite surprised to see clippy complain about it.